cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/KafkaLogAppender.java

```diff
@@ -206,14 +206,23 @@ private void publishMessages(List<KeyedMessage<String, byte[]>> buffer,
 
       if (blockForMessage) {
         blockingThread = Thread.currentThread();
+        LogMessage message = null;
         try {
           if (isRunning()) {
-            buffer.add(createKeyedMessage(messageQueue.take()));
+            message = messageQueue.take();
+            buffer.add(createKeyedMessage(message));
             maxBufferSize--;
           }
         } catch (InterruptedException e) {
           // just ignore and keep going. This happen when this publisher is getting shutdown, but we still want
           // to publish all pending messages.
+        } catch (Throwable t) {
+          if (message == null) {
+            contextAware.addError("Got exception while taking message from queue", t);
+          } else {
+            contextAware.addError("Got exception while adding log message to queue: "
```

Comment:
Not sure whether there are any runtime exceptions from this method. Saw the same try-catch block in previous implementation, so I think this may be necessary to prevent skipping other messages when one message fails to be encoded